### ISSUE TITLE
Incorrect documentation: Mongo discriminator sample triggers Mongoose model overwrite error

### ISSUE THREAD

**Issue by inn0vative1** [nestjs/nest#6517]
## Bug Report
Example in documentation throws an unrecoverable error when registering discriminators for a schema.
The example is shown in the Mongo -> Discriminators section (https://docs.nestjs.com/techniques/mongodb#discriminators)

## Current behavior
Copy the example code out of the documentation. . 
Or clone this repo which copies from the sample: https://github.com/inn0vative1/nestjs-events-sample

Run `npm install` and `npm start`

Application throws error:

 `[ExceptionHandler] Cannot overwrite 'ClickedLinkEvent' model once compiled.`

Stack trace: 
```
OverwriteModelError: Cannot overwrite `ClickedLinkEvent` model once compiled.
    at Function.Model.discriminator (/***/nestjs-events-sample-master/node_modules/mongoose/lib/model.js:1135:11)
    at /***/nestjs-events-sample-master/node_modules/@nestjs/mongoose/dist/mongoose.providers.js:15:56
    at Array.forEach (<anonymous>)
    at addDiscriminators (/***/nestjs-events-sample-master/node_modules/@nestjs/mongoose/dist/mongoose.providers.js:15:20)
    at InstanceWrapper.useFactory [as metatype] (/***/nestjs-events-sample-master/node_modules/@nestjs/mongoose/dist/mongoose.providers.js:25:17)
    at Injector.instantiateClass (/***/nestjs-events-sample-master/node_modules/@nestjs/core/injector/injector.js:289:55)
    at callback (/***/nestjs-events-sample-master/node_modules/@nestjs/core/injector/injector.js:42:41)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Injector.resolveConstructorParams (/***/nestjs-events-sample-master/node_modules/@nestjs/core/injector/injector.js:114:24)
    at async Injector.loadInstance (/***/nestjs-events-sample-master/node_modules/@nestjs/core/injector/injector.js:46:9)
    at async Injector.loadProvider (/***/nestjs-events-sample-master/node_modules/@nestjs/core/injector/injector.js:68:9)
    at async Promise.all (index 5)
    at async InstanceLoader.createInstancesOfProviders (/***/nestjs-events-sample-master/node_modules/@nestjs/core/injector/instance-loader.js:43:9)
    at async /***/nestjs-events-sample-master/node_modules/@nestjs/core/injector/instance-loader.js:28:13
    at async Promise.all (index 5)
    at async InstanceLoader.createInstances (/***/nestjs-events-sample-master/node_modules/@nestjs/core/injector/instance-loader.js:27:9)
```

## Environment

<pre><code>
- Nest version: 7.6.13
- Node version: 14.14
- Platform:  Mac
</code></pre>

## Other Info
The underlying error is a Mongoose error, but it appears the registering discriminators in NestJs following the sample code triggers this error. https://mongoosejs.com/docs/faq.html#overwrite-model-error

**Comment by jmcdo29** [comment]
There's a PR in place to fix the mongoose discriminators at nestjs/mongoose#692

**Comment by kamilmysliwiec** [comment]
Fixed in @nestjs/mongoose@7.2.4
